tests/semantic_analyser.cpp

```diff
@@ -956,13 +947,13 @@ kprobe:f { @y[(3, 4, 5)] = 5; delete(@y, (1, 2)); }
 )");
 
   test_error("kprobe:f { @y[1] = 2; delete(@y); }", R"(
-stdin:1:23-32: ERROR: delete() expects a map for the first argument and a key for the second argument e.g. `delete(@my_map, 1);`
+stdin:1:23-32: ERROR: delete() expects a map without explicit keys (scalar map)
```

Comment:
this is just wrong. it should be something like "delete(), when called with 1 argument, expects a map without explicit keys (scalar map)."

Reply:
The error message is for the current form. Per below, can I do `call to delete() expects a map without explicit keys ...`

Reply:
I've backported this (pass wise) in the same format used by the semantic analyser (e.g. `requires 1 or 2 arguments`) as a special case for `delete`.